tools/convert_datasets/mot2reid.py

```diff
@@ -0,0 +1,169 @@
+# This script converts MOT dataset into ReID dataset.
+# Offical website of the MOT dataset: https://motchallenge.net/
+#
+# Label format of MOT dataset:
+#   GTs:
+#       <frame_id> # starts from 1 but COCO style starts from 0,
+#       <instance_id>, <x1>, <y1>, <w>, <h>,
+#       <conf> # conf is annotated as 0 if the object is ignored,
+#       <class_id>, <visibility>
+#
+#   DETs and Results:
+#       <frame_id>, <instance_id>, <x1>, <y1>, <w>, <h>, <conf>,
+#       <x>, <y>, <z> # for 3D objects
+#
+# Classes in MOT:
+#   1: 'pedestrian'
+#   2: 'person on vehicle'
+#   3: 'car'
+#   4: 'bicycle'
+#   5: 'motorbike'
+#   6: 'non motorized vehicle'
+#   7: 'static person'
+#   8: 'distractor'
+#   9: 'occluder'
+#   10: 'occluder on the ground',
+#   11: 'occluder full'
+#   12: 'reflection'
+#
+#   USELESS classes and IGNORES classes will not be selected
+#   into the dataset for reid model training.
+import argparse
+import os
+import os.path as osp
+import random
+
+import mmcv
+import numpy as np
+from tqdm import tqdm
+
+USELESS = [3, 4, 5, 6, 9, 10, 11]
+IGNORES = [2, 7, 8, 12, 13]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Convert MOT label and detections to COCO-VID format.')
+    parser.add_argument('-i', '--input', help='path of MOT data')
+    parser.add_argument(
+        '-o', '--output', help='path to save coco formatted label file')
+    parser.add_argument(
+        '--val-split', type=float, default=0.2, help='path of MOT data')
+    parser.add_argument(
+        '--vis-threshold',
+        type=float,
+        default=0.3,
+        help='threshold of visibility for each person')
+    parser.add_argument(
+        '--min-per-person',
+        type=int,
+        default=8,
+        help='minimum number of images for each person')
+    parser.add_argument(
+        '--max-per-person',
+        type=int,
+        default=1000,
+        help='maxmum number of images for each person')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    if not osp.exists(args.output):
+        os.makedirs(args.output)
+    elif os.listdir(args.output):
+        raise OSError(f'Directory must empty: \'{args.output}\'')
+
+    in_folder = osp.join(args.input, 'train')
+    video_names = os.listdir(in_folder)
+    sorted(video_names)
+    if 'MOT17' in in_folder:
+        video_names = [
+            video_name for video_name in video_names if 'FRCNN' in video_name
+        ]
+    for video_name in tqdm(video_names):
+        # load video infos
+        video_folder = osp.join(in_folder, video_name)
+        infos = mmcv.list_from_file(f'{video_folder}/seqinfo.ini')
+        # video-level infos
+        assert video_name == infos[1].strip().split('=')[1]
+        raw_img_folder = infos[2].strip().split('=')[1]
+        raw_img_names = os.listdir(f'{video_folder}/{raw_img_folder}')
+        raw_img_names = sorted(raw_img_names)
```

Comment:
Why do you need to sort the images?

Reply:
Because we have to read the pictures in the order of the frames. For example, the `raw_img_names[0]` should be `000001.jpg`